lib/shimmer.dart

```diff
@@ -194,37 +222,41 @@ class _ShimmerFilter extends RenderProxyBox {
     if (child != null) {
       assert(needsCompositing);
 
-      final width = child.size.width;
-      final height = child.size.height;
-      Rect rect;
-      double dx, dy;
-      if (_direction == ShimmerDirection.rtl) {
-        dx = _offset(width, -width, _percent);
-        dy = 0.0;
-        rect = Rect.fromLTWH(offset.dx - width, offset.dy, 3 * width, height);
-      } else if (_direction == ShimmerDirection.ttb) {
-        dx = 0.0;
-        dy = _offset(-height, height, _percent);
-        rect = Rect.fromLTWH(offset.dx, offset.dy - height, width, 3 * height);
-      } else if (_direction == ShimmerDirection.btt) {
-        dx = 0.0;
-        dy = _offset(height, -height, _percent);
-        rect = Rect.fromLTWH(offset.dx, offset.dy - height, width, 3 * height);
-      } else {
-        dx = _offset(-width, width, _percent);
-        dy = 0.0;
-        rect = Rect.fromLTWH(offset.dx - width, offset.dy, 3 * width, height);
-      }
-      if (_rect != rect) {
-        _gradientPaint.shader = _gradient.createShader(rect);
-        _rect = rect;
-      }
-
       context.canvas.saveLayer(offset & child.size, _clearPaint);
       context.paintChild(child, offset);
-      context.canvas.translate(dx, dy);
-      context.canvas.drawRect(rect, _gradientPaint);
-      context.canvas.restore();
+
+      if (enabled) {
+        final width = child.size.width;
+        final height = child.size.height;
+        Rect rect;
+        double dx, dy;
+        if (_direction == ShimmerDirection.rtl) {
+          dx = _offset(width, -width, _percent);
+          dy = 0.0;
+          rect = Rect.fromLTWH(offset.dx - width, offset.dy, 3 * width, height);
+        } else if (_direction == ShimmerDirection.ttb) {
+          dx = 0.0;
+          dy = _offset(-height, height, _percent);
+          rect =
+              Rect.fromLTWH(offset.dx, offset.dy - height, width, 3 * height);
+        } else if (_direction == ShimmerDirection.btt) {
+          dx = 0.0;
+          dy = _offset(height, -height, _percent);
+          rect =
+              Rect.fromLTWH(offset.dx, offset.dy - height, width, 3 * height);
+        } else {
+          dx = _offset(-width, width, _percent);
+          dy = 0.0;
+          rect = Rect.fromLTWH(offset.dx - width, offset.dy, 3 * width, height);
+        }
+        if (_rect != rect) {
+          _gradientPaint.shader = _gradient.createShader(rect);
+          _rect = rect;
+        }
+        context.canvas.translate(dx, dy);
+        context.canvas.drawRect(rect, _gradientPaint);
+        context.canvas.restore();
```

Comment:
If `enabled` is `false`, `canvas.restore()` is never called. You think it's ok?

Reply:
Oh I am not really an expert on canvas. But I think we could add it, because we're calling canvas.saveLayer()

Reply:
I had a look at `saveLayer` document, we should call `restore` at the end of our painting manipulation codes. `saveLayer` will put the current transform and clipping context on the stack, `restore` will pop the save stack and apply the paint to the group.

Reply:
I just tried it out, it turns out that the `paint` method is never called at all when `enabled` is set to `false`. So we can ignore `if` clause inside `paint` method.

Reply:
Ok i tested it a bit and I found a situation where not calling canvas.restore() caused a rendering issue.